### PR TITLE
Use docker-machine ssh on version 0.5.* of docker-machine

### PR DESCRIPTION
**Pull request by huggsboson**

### CODE REVIEW

**Comment by huggsboson** [comment]
I get non-fatal, but annoying errors about tcp and bad certs if I use your ssh command as opposed to docker-machine ssh on 0.5.1

**Comment by huggsboson** [comment]
The specific error is:
channel 2: open failed: connect failed: Connection refused

**Comment by huggsboson** [comment]
again the scripts still work it just spews that while it's "Waiting for Kubernetes cluster to become available..."

**Comment by vyshane** [comment]
I think that we should just switch to `docker-machine ssh` for port forwarding since it now supports this out of the box. Can you update the PR to do so? We also need to remove the port forward upon kube-down.

**Comment by huggsboson** [comment]
I've added the killing to kube-down.   Sadly the old ssh 1:1 correspondance with the command we ran is gone, so I have to detect something different than what we run, it now looks like:
/usr/bin/ssh -o PasswordAuthentication=no -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=no docker@127.0.0.1 -i /Users/jhuffaker/.docker/machine/machines/default1/id_rsa -p 51174 -f -N -L 8080:localhost:8080

I could maybe expand my pgrep and pkill match to include docker@127.0.0.1 to make it tighter.

I'm struggling a bit to think through the right behavior in a lot of these cases.

**Comment by huggsboson** [comment]
I tested the stuff that is up by running kube-up and kube-down both with an existing ssh session forwarding 8080 and without.  In the case where there isn't an existing one it starts up and shutsdown (and properly kills the forward).

In the case where there is an existing forward it maintains the existing behavior of not starting a new one, and not killing it on kube-down.  But basically kubectl won't work and it will hang waiting for kube to become available.

If we want to change that second behavior I can just move to killing all ssh sessions that are holding 8080.

**Comment by huggsboson** [comment]
thoughts?  The current ssh command gives a bunch fun things around needing to type yes to accept the cert and other issues with known hosts as well as the connect failed I mentioned above.

File: scripts/docker-machine-port-forwarding.sh [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
-    if [ -z "$existing_forward" ]; then
-        eval $forward_port_command
+    local port=$1
+    local machine=$(active_machine)
+
+    if [ -n "$machine" ]; then
+        if ! pgrep -f "ssh.*$port:localhost" > /dev/null; then
+            docker-machine ssh "$machine" -f -N -L "$port:localhost:$port"
+        else
+            echo "An ssh tunnel on port $port already exists.  The kubernetes cluster won't be reachable from local kubectl."
```
Comment (vyshane) [inline scripts/docker-machine-port-forwarding.sh]: Can we change the error message to read "Did not set up port forwarding to the Docker machine: An ssh tunnel on port $port already exists. The kubernetes cluster may not be reachable from local kubectl."

**Comment by vyshane** [comment]
@huggsboson thanks for that. Just had a small comment about the error message. Otherwise LGTM, happy to merge it in.

**Comment by huggsboson** [comment]
Cool, much better error text.  Patchset updated.

**Pull request merged**